### ISSUE TITLE
i got bad clustering score for cifar-10 in resnet 50

### ISSUE THREAD

**Issue by gargatik** [wgcban/mix-bt#1]
@wgcban 
Cluster Centers:
[[0.00748559 0.00252423 0.02586649 ... 0.00637615 0.0058655  0.00041923]
 [0.00362701 0.00321243 0.01586772 ... 0.0027485  0.00394693 0.01120272]
 [0.00392837 0.00411785 0.01762331 ... 0.00431684 0.00594873 0.0005309 ]
 ...
 [0.00623401 0.00322232 0.03249142 ... 0.00796974 0.00280931 0.000397  ]
 [0.00561143 0.00398928 0.02272538 ... 0.00499028 0.00987207 0.00204776]
 [0.00240571 0.00451606 0.01574576 ... 0.00336538 0.00481124 0.0005551 ]]
Adjusted Rand Index (Train): -2.352540539571238e-05
Normalized Mutual Information (Train): 0.0024080991363850815
Silhouette Score (Train): 0.016357887536287308
Homogeneity (Train): 0.001759865775481539
Completeness (Train): 0.0038123494767823784
V-measure (Train): 0.0024080991363850815

Hi, thanks for the great method. i tried to use your mehtod to find clustering scores. Hoowever si do not get good scores despite high accracy. can you tell me what is the issue?

i tried both feature and out in line 118 from main.py to get my clustering score with the help fo the given checkpoints for resnet-50

**Comment by wgcban** [comment]
@gargatik Can you please also check the k-NN accuracy to verify everything else is correct? 
Can you get lower clustering score despite higher k-NN accuracy? Is it possible?

**Comment by wgcban** [comment]
@gargatik I can also check from my side. Can you please post the code you used to compute clustering scores? Thanks.

**Comment by gargatik** [comment]
@wgcban I have identified the problem. It appears that there might be issues with the majority of the pretrained checkpoints provided. However, upon training the model by myself and applying clustering techniques, I noticed a significant and dramatic improvement in the scores.

Can you update the correct pretrained checkpoints?

**Comment by gargatik** [comment]
@wgcban Has your paper been accepted for publication or is it still in the preprint stage? It would be helpful to me if you can provide me updated citation if its already accepted somewhere

**Comment by wgcban** [comment]
@gargatik thanks for confirming. I am happy that  you observed a good score. I will check with pre-trained checkpoints again. So far I do not observe any issues with them. Please check again. My paper still in the pre-print stage. Thanks.

**Comment by wgcban** [comment]
@gargatik Note that the checkpoints here provided are only the resnet. It does not include the classification head weights.

**Comment by gargatik** [comment]
> @gargatik Note that the checkpoints here provided are only the resnet. It does not include the classification head weights.

@wgcban 
Will it affect the features and output in line 118 of main.py to calculate the clustering score? I observed a significant difference when using ResNet-50 checkpoints provided by you or those trained by myself.

**Comment by wgcban** [comment]
@gargatik it will not affect the `features`, but `outputs` -- because `features` are the output from the `reset50` backbone while `outputs` goes through additional `projector` network whose weights are not saved in the provided checkpoints.

**Comment by wgcban** [comment]
For CIFARs and other small datasets, the provided checkpoints include both `backbone` and `projector`. But for the `imagenet`, it only includes the backbone. 
If you initialize both `backbone` and the `projector`, it should not affect `features` and `outputs` for small datasets. Did you load the pre-trained model correctly: 
`self.load_state_dict(torch.load(pretrained_path, map_location='cpu'), strict=True)`
Thanks

**Comment by wgcban** [comment]
I think this issue has been resolved, hence closing the issue.
